tests/data/test_md_report_only_failed_tests.md

```diff
@@ -0,0 +1,101 @@
+| ------------------| ----------- | ------------- | ------------- | ------------- | ----------- | -------------------| ------------------|
+| DC1-SPINE1 | 44 | 13 | 6 | 24 | 1 | MLAG, OSPF, VXLAN | AAA, BFD, BGP, Connectivity, Interfaces, Multicast, Routing, SNMP, STP, Software, System, VLAN |
+| DC1-LEAF1A | 45 | 18 | 2 | 24 | 1 | OSPF | AAA, BFD, BGP, Connectivity, Interfaces, Multicast, Routing, SNMP, STP, Software, System, VLAN |
+
+### Summary Totals Per Category
+
+| Test Category | Total Tests | Tests Success | Tests Skipped | Tests Failure | Tests Error |
+| ------------- | ----------- | ------------- | ------------- | ------------- | ----------- |
+| AAA | 2 | 0 | 0 | 2 | 0 |
+| BFD | 6 | 2 | 0 | 4 | 0 |
+| BGP | 22 | 0 | 0 | 22 | 0 |
+| Connectivity | 3 | 0 | 0 | 1 | 2 |
+| Interfaces | 10 | 6 | 0 | 4 | 0 |
+| MLAG | 4 | 2 | 2 | 0 | 0 |
+| Multicast | 4 | 2 | 0 | 2 | 0 |
+| OSPF | 4 | 0 | 4 | 0 | 0 |
+| Routing | 6 | 3 | 0 | 3 | 0 |
+| SNMP | 2 | 0 | 0 | 2 | 0 |
+| STP | 2 | 0 | 0 | 2 | 0 |
+| Security | 4 | 4 | 0 | 0 | 0 |
+| Software | 2 | 0 | 0 | 2 | 0 |
+| System | 12 | 10 | 0 | 2 | 0 |
+| VLAN | 2 | 0 | 0 | 2 | 0 |
+| VXLAN | 4 | 2 | 2 | 0 | 0 |
+
+## Failed Test Results Summary
+
+| Device Under Test | Categories | Test | Description | Custom Field | Result | Messages |
+| ----------------- | ---------- | ---- | ----------- | ------------ | ------ | -------- |
+| DC1-LEAF1A | BFD | VerifyBFDPeersIntervals | Verifies the timers of the IPv4 BFD peers in the specified VRF. | - | failure | Following BFD peers are not configured or timers are not correct: {'192.0.255.8': {'default': 'Not Configured'}, '192.0.255.7': {'default': 'Not Configured'}} |
+| DC1-LEAF1A | BFD | VerifyBFDSpecificPeers | Verifies the IPv4 BFD peer's sessions and remote disc in the specified VRF. | - | failure | Following BFD peers are not configured, status is not up or remote disc is zero: {'192.0.255.8': {'default': 'Not Configured'}, '192.0.255.7': {'default': 'Not Configured'}} |
+| DC1-LEAF1A | BGP | VerifyBGPAdvCommunities | Verifies the advertised communities of a BGP peer. | - | failure | Following BGP peers are not configured or advertised communities are not standard, extended, and large: {'bgp_peers': {'172.30.11.17': {'default': {'status': 'Not configured'}}, '172.30.11.21': {'default': {'status': 'Not configured'}}}} |
+| DC1-LEAF1A | BGP | VerifyBGPExchangedRoutes | Verifies the advertised and received routes of BGP peers. | - | failure | Following BGP peers are not found or routes are not exchanged properly: {'bgp_peers': {'172.30.255.5': {'default': 'Not configured'}, '172.30.255.1': {'default': 'Not configured'}}} |
+| DC1-LEAF1A | BGP | VerifyBGPPeerASNCap | Verifies the four octet asn capabilities of a BGP peer. | - | failure | Following BGP peer four octet asn capabilities are not found or not ok: {'bgp_peers': {'172.30.11.1': {'default': {'status': 'Not configured'}}}} |
```

Comment:
do we need48 tests in tests data?

Reply:
I wanted to have a little bit of everything and I don't think it impacts performance that much.

Reply:
well .. what it impacts is maintainability - if we deprecate tests for instance we need to come back here and modify this file - which is big and which we will probably forget

Reply:
Reduced the amount of tests in fb2139a3fd7763022aec4746db3a350aea9404c3